### ISSUE TITLE
Authenticator component returns idle in authenticated state [2]

### ISSUE THREAD

**Issue by zvictor** [aws-amplify/amplify-ui#4905]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Next.js

### What browsers are you seeing the problem on?

Chrome, Firefox

### Please describe your bug.

Note: _Similar issues have been reported before, but I believe there are important nuances on this one._

While using the `<Authenticator>` component (as a wrapper of my App) to setup Amplify in the new Nextjs app router, I could notice that logged in users would sometimes be mistakenly redirected to `/sign-in` but been shown a 404 page instead of the signInUp form.

I did some debugging and could notice that `useAuthenticator` is returning `idle` as `route`.

https://github.com/aws-amplify/amplify-ui/blob/c021e4900f96297c36fed74cd01bc29eb7c5827d/packages/react/src/components/Authenticator/Router/Router.tsx#L61

@ErikCH [commented on a similar issue](https://github.com/aws-amplify/amplify-ui/issues/1497#issuecomment-1112457739):

> On refresh the route will temporarily be in a setup or idle state before it transitions to an authenticated state. If you check route as soon as the page loads, and redirect somewhere while it's in the idle or setup state, that could be an issue.
> 
> ...
> 
> In this scenario if someone goes to an authenticated route, and it's an idle or setup state then we redirect back to /login. Which will then by that time see the user is authenticated and will re-route back to the authenticated page.

Related to: #1332 #1497 #3727

### What's the expected behaviour?

1. Logged in users should not be redirected to `/sign-in`
2. Logged in users should not see a 404 page when visiting `/sign-in`

### Help us reproduce the bug!

1. Start a new next.js project
2. Add `Authenticator` to `/app/providers.tsx`, as shown below:

```tsx
'use client'

import { Amplify } from 'aws-amplify'
import { Authenticator } from '@aws-amplify/ui-react'
import * as amplifyConfig from '@/modules/auth/config'
import { ThemeProvider as NextThemesProvider } from 'next-themes'
import '@aws-amplify/ui-react/styles.css'

Amplify.configure(amplifyConfig, { ssr: true })

export function Providers({ children }: { children: React.ReactNode }) {
  return (
      <NextThemesProvider attribute="class" defaultTheme="system">
        <Authenticator variation="modal">{children}</Authenticator>
      </NextThemesProvider>
  )
}
```

Once that is done you should notice the problem reported as long as you try navigating and reloading the pages while logged in.

### Code Snippet

```json
  "dependencies": {
    "@aws-amplify/adapter-nextjs": "1.0.10",
    "@aws-amplify/auth": "6.0.10",
    "@aws-amplify/ui-react": "6.1.0",
    "aws-amplify": "6.0.10",
    "next": "14.0.4",
    "next-themes": "0.2.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
  }
```

**Comment by reesscot** [comment]
Hi @zvictor,
I'm sorry you're having trouble with the `Authenticator`. Can you please provide more details on how you are doing the redirect to your `/sign-in` page? The `Authenticator` is a Client Component and doesn't have any control over routing in Next.js App Router. I'm unclear what you're trying to accomplish using the useAuthenticator hook, but that will not work in a server component. The best way to handle routing is to use middleware to do the /sign-in redirect on the server side. You may want to take a look at this guide:
https://docs.amplify.aws/gen2/start/quickstart/nextjs-app-router/

**Comment by zvictor** [comment]
Oh, you are right! I didn't remember setting up `middleware.ts` in my app and I was really struggling to understand why there was a redirection happening. I thought there was some magic happening inside `Authenticator`, but couldn't find it in the source code 🤦 

Anyway, the content of my `middleware.ts` is the same recommended by the [Server-Side Rendering guide](https://docs.amplify.aws/react/build-a-backend/server-side-rendering/#authentication-with-nextjs-server-side-runtime), verbatim.

**Comment by reesscot** [comment]
@zvictor Can you clarify how you are trying to use `useAuthenticator` in your situation? It's not recommended to wrap your application in Authenticator when using the Next.js App Router, it would be better to render it on your `/sign-in` page instead and use middleware to protect your routers.

**Comment by reesscot** [comment]
@zvictor Following up here to say that our [Next.js App Router (Server Components)](https://docs.amplify.aws/gen2/start/quickstart/nextjs-app-router-server-components/) guide has been published. 

Did you have any further questions?

**Comment by reesscot** [comment]
Closing this out as we haven't heard back. Please open a new issue if you are still having trouble with App Router.
